### DISCUSSION TITLE
Python examples cannot find raisimPy

### DISCUSSION THREAD

**Discussion by waleedbkhalid** [raisimTech/raisimLib#512] [General]
![Screenshot from 2023-10-05 16-20-07](https://github.com/raisimTech/raisimLib/assets/89767563/c60d3dd5-6d2c-4d75-8c2e-183be3a081cc)

I have been running into this issue. I have followed the instructions carefully and even tried manually giving the raisimPy library path to my conda environment.

Any ldeas on how to fix this would be very helpful.

**Comment by jhwangbo** [comment]
Can you type ``echo $LD_LIBRARY_PATH`` and paste the output?

**Comment by waleedbkhalid** [comment]
I am not near my machine right now, so I will do that later.

However, I did have these two paths in my bashrc file.

But can you tell me about -DCMAKE_PREFIX_PATH=$WORKSPACE/raisim/linux. this part of the instruction

When do i add this during installation.

I will provide the other requested information, as soon as I get near my machine.

**Comment by jhwangbo** [comment]
As the error states, the python finds ``raisimPy`` but not ``libraisim``. The mechanism to find the library is these two lines
export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$WORKSPACE/raisim/linux/lib
export PYTHONPATH=$PYTHONPATH:$WORKSPACE/raisim/linux/lib
This adds raisim libraries to the environment variable "LD_LIBRARY_PATH". So if this is done correctly, you should see the raisim path after calling ``echo $LD_LIBRARY_PATH``

**Reply by waleedbkhalid** [reply]
I see. But I also have another question

why can't it be this way

export LD_LIBRARY_PATH=$WORKSPACE/raisim/linux/lib
export PYTHONPATH=$WORKSPACE/raisim/linux/lib

why do we need to have an extra $LD_LIBRARY_PATH: and $PYTHONPATH in the below statements, which I copied from above.

export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$WORKSPACE/raisim/linux/lib
export PYTHONPATH=$PYTHONPATH:$WORKSPACE/raisim/linux/lib

**Reply by jhwangbo** [reply]
Because the former will overwrite your existing path list. We want to append to your existing list so that we don't mess up with your other projects/libraries.

**Reply by waleedbkhalid** [reply]
Right. I actually am in the process of reinstalling things now to make sure I follow everything closely.

But i just realized another thing

The exact statements suggest that the raisim folder is in the workspace folder that i made. However, when i clone the repo, i clone it in the workspace folder.

the cloned folder is downloaded as raisimLib and the remaining path is in raisimlib

So shouldnt this be 
export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$WORKSPACE/raisimlib/raisim/linux/lib
export PYTHONPATH=$PYTHONPATH:$WORKSPACE/raisimlib/raisim/linux/lib

**Reply by jhwangbo** [reply]
yes, that is the correct paht. I'll fix the doc

**Reply by waleedbkhalid** [reply]
Cool

and lastly, can you also explain
![Screenshot from 2023-10-05 23-53-58](https://github.com/raisimTech/raisimLib/assets/89767563/34d80099-a047-4ca1-8ead-b2a346cc2d8c)
The cmake instructions in this section.

when exactly do i run this during the setup for raisimPy and what is the exact command to add this to my cmake.

**Reply by jhwangbo** [reply]
If you have another CMake project that needs Raisim, you should add ``-DCMAKE_PREFIX_PATH=$WORKSPACE/raisim/linux`` when you run your project's cmake

**Comment by waleedbkhalid** [comment]
cmake -DCMAKE_PREFIX_PATH=$WORKSPACE/raisim/linux ..

Chat gpt suggested this

and i did that in the raisimPy instructions, before I ran the cmake command

But i think the reason it didnt work was that i did not pay attention to adding raisimLib here.

do u think that could be the reason? I am about to run it again and see.

**Comment by waleedbkhalid** [comment]
![image](https://github.com/raisimTech/raisimLib/assets/89767563/60d1e57c-dd43-4727-bd78-4b1510079f99)
Nope i still ran into the same issue.

![image](https://github.com/raisimTech/raisimLib/assets/89767563/cd27ff47-bdb1-4e0c-ac12-e3359d762946)
this is what i did and then ran the make install -j4 command.

**Comment by waleedbkhalid** [comment]
![image](https://github.com/raisimTech/raisimLib/assets/89767563/dee0a696-b0bc-48fa-8a00-ac4f4dffec63)
This is the result of the echo command u asked me to run

**Reply by jhwangbo** [reply]
Please do not copy paste a screenshot, that is not searchable. Your image says ``raisimlib`` and your cmake outputs say ``raisimLib``

**Reply by waleedbkhalid** [reply]
Apologies. I will be careful. Yes I just realized that and fixed it. 

Thank you for your time. And thanks for being so prompt. For future scenarios, how do you want me to post. I would like a little elaboration on how do you want me to post the screenshot so that it is searchable. or Do you want me to copy paste the error message?
